XML/syntax_test_xml.xml

```diff
@@ -201,6 +222,26 @@
      >
 <!-- ^ text - punctuation - illegal -->
 
+     <NS:TAGNAME XMLNS:NS="URI">
+<!-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.tag -->
+<!--                            ^ - meta.tag -->
+<!-- ^ punctuation.definition.tag.begin -->
+<!--  ^^ entity.name.tag.namespace -->
+<!--    ^ punctuation.separator.namespace -->
+<!--     ^^^^^^^ entity.name.tag.localname -->
+<!--            ^ - entity -->
+<!--             ^^^^^ entity.other.attribute-name.namespace -->
```

Comment:
Note that `xmlns` must be lowercase to be treated as a namespace declaration, so this is not defining a namespace with prefix `NS`, so this node is therefore only valid if the `NS` prefix has been defined previously in document order. As the syntax definition doesn't scope the attribute differently, there is actually nothing wrong with this syntax test, except potential confusion in future if the syntax definition gets changed to scope namespace declarations differently.

> Error on->  Line :2 Column :16
Message :Namespace prefix XMLNS for NS on TAGNAME is not defined

> Error on->  Line :2 Column :16
Message :Namespace prefix NS on TAGNAME is not defined